presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/RFC2822FormatFunctions.java

```diff
@@ -0,0 +1,89 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+import java.util.function.Function;
+
+import static io.prestosql.spi.type.DateTimeEncoding.unpackMillisUtc;
+import static io.prestosql.spi.type.DateTimeEncoding.unpackZoneKey;
+import static java.time.ZoneOffset.UTC;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public final class RFC2822FormatFunctions
+{
+    private static final DateTimeFormatter RFC_FORMATTER = DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss Z yyyy").withLocale(Locale.ENGLISH);
+
+    private RFC2822FormatFunctions() {}
+
+    public static Function<Long, String> formatDateFunc()
+    {
+        return millis -> RFC_FORMATTER.format(
+                ZonedDateTime.of(
+                        LocalDate.ofEpochDay(MILLISECONDS.toDays(millis)),
+                        LocalTime.MIDNIGHT,
+                        UTC));
+    }
+
+    public static Function<Long, String> formatTimeFunc()
+    {
+        return millis -> RFC_FORMATTER.format(
+                ZonedDateTime.of(
+                        LocalDate.EPOCH,
+                        localTimeOfEpochMillis(millis),
+                        UTC));
+    }
+
+    public static Function<Long, String> formatTimeWithTZFunc()
+    {
+        return encodedMillisWithTZ -> RFC_FORMATTER.format(
+                ZonedDateTime.of(
+                        LocalDate.EPOCH,
+                        localTimeOfEpochMillis(unpackMillisUtc(encodedMillisWithTZ)),
+                        unpackZoneKey(encodedMillisWithTZ).getZoneId()));
+    }
+
+    public static Function<Long, String> formatTimestampFunc()
+    {
+        return millis -> RFC_FORMATTER.format(
+                ZonedDateTime.of(
+                        localDateTimeOfEpochMillis(millis),
+                        UTC));
+    }
+
+    public static Function<Long, String> formatTimestampWithTZFunc()
+    {
+        return encodedMillisWithTZ -> RFC_FORMATTER.format(
+                ZonedDateTime.of(
+                        localDateTimeOfEpochMillis(unpackMillisUtc(encodedMillisWithTZ)),
+                        unpackZoneKey(encodedMillisWithTZ).getZoneId()));
+    }
+
+    private static LocalDateTime localDateTimeOfEpochMillis(long epochMillis)
```

Comment:
Perhaps this would warrant being public in `JsonFormatFunctions`.  It's being used in `ISO8601FormatFunctions`, as well.